### PR TITLE
Candidates photometric annotations filter

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR adds a filter on photometric annotations for candidates.

### CODE REVIEW

**Review by guynir42 — APPROVED** [review]
Good! I have a couple of small suggestions.

File: skyportal/handlers/api/candidate.py [hunk 1]
```diff
@@ -603,6 +673,22 @@ def get(self, obj_id=None):
...
+            "photometryAnnotationsFilterOrigin", None
+        )
+        photometry_annotations_filter_after = self.get_query_argument(
+            'photometryAnnotationsFilterAfter', None
+        )
+        photometry_annotations_filter_before = self.get_query_argument(
+            'photometryAnnotationsFilterBefore', None
+        )
+        photometry_annotations_filter_threshold = self.get_query_argument(
+            'photometryAnnotationsFilterThreshold', 1
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/candidate.py]: Good idea.

File: skyportal/handlers/api/candidate.py [hunk 2]
```diff
@@ -909,6 +995,122 @@ def get(self, obj_id=None):
...
+            if photometry_annotations_filter is not None:
+                if (
+                    isinstance(photometry_annotations_filter, str)
+                    and "," in photometry_annotations_filter
+                ):
+                    photometry_annotations_filter = [
+                        c.strip() for c in photometry_annotations_filter.split(",")
+                    ]
+                elif isinstance(photometry_annotations_filter, str):
+                    photometry_annotations_filter = [photometry_annotations_filter]
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/candidate.py]: Yup.

File: skyportal/handlers/api/candidate.py [hunk 3]
```diff
@@ -909,6 +995,122 @@ def get(self, obj_id=None):
...
+                ):
+                    photometry_annotations_filter_origin = [
+                        c.strip()
+                        for c in photometry_annotations_filter_origin.split(",")
+                    ]
+                elif isinstance(photometry_annotations_filter_origin, str):
+                    photometry_annotations_filter_origin = [
+                        photometry_annotations_filter_origin
+                    ]
+                else:
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/candidate.py]: Yup.

**Pull request merged**